File: Java/src/main/java/Reservation.java
import java.util.List;

public class Reservation {

    public final TrainId trainId;
    public final String bookingReference;
    public final List<Seat> seats;

    public Reservation(TrainId trainId, String bookingReference, List<Seat> seats) {
        this.trainId = trainId;
        this.bookingReference = bookingReference;
        this.seats = seats;
    }

}

File: csharp/src/Reservation.cs
using System.Collections.Generic;

namespace TrainKata
{
    public struct Reservation
    {
        public TrainId TrainId { get; }

        public BookingReference BookingReference { get; }

        public IEnumerable<SeatId> Seats { get; }

        public Reservation(TrainId trainId, BookingReference bookingReference, IEnumerable<SeatId> seats)
        {
            TrainId = trainId;
            BookingReference = bookingReference;
            Seats = seats;
        }
    }
}